cedar-policy-validator/src/level_validate.rs

```diff
@@ -77,8 +79,8 @@ impl Validator {
     }
 
     fn min(
-        v: impl IntoIterator<Item = (EntityDerefLevel, Option<EntityDerefLevelViolation>)>,
-    ) -> (EntityDerefLevel, Option<EntityDerefLevelViolation>) {
+        v: impl IntoIterator<Item = (EntityDerefLevel, Option<ValidationError>)>,
+    ) -> (EntityDerefLevel, Option<ValidationError>) {
```

Comment:
I don't understand why the changes in this file are needed.

Reply:
So the issue here is that we need to handle the case of an error node. It doesn't entirely make sense to return an `EntityDerefLevelViolation` in the case of an error. I discussed with John, and he suggested that this might be a better approach so that we could use `ValidationError::InternalInvariantViolation` here 

Definitely open to other suggestions!